devito/nodes.py

```diff
@@ -350,18 +358,28 @@ def ccode(self):
 
         # For reverse dimensions flip loop bounds
         if self.reverse:
-            loop_init = c.InlineInitializer(c.Value("int", self.index),
-                                            ccode('%s - 1' % end))
+            loop_init = 'int %s = %s' % (self.index, ccode('%s - 1' % end))
             loop_cond = '%s >= %s' % (self.index, ccode(start))
             loop_inc = '%s -= %s' % (self.index, self.limits[2])
         else:
-            loop_init = c.InlineInitializer(c.Value("int", self.index), ccode(start))
+            loop_init = 'int %s = %s' % (self.index, ccode(start))
             loop_cond = '%s < %s' % (self.index, ccode(end))
             loop_inc = '%s += %s' % (self.index, self.limits[2])
 
+        # Append unbounded indices, if any
+        if self.uindices:
+            uinit = ['%s = %s' % (i.index, ccode(i.start)) for i in self.uindices]
+            loop_init = c.Line(', '.join([loop_init] + uinit))
+            ustep = ['%s = %s' % (i.index, ccode(i.step)) for i in self.uindices]
+            loop_inc = c.Line(', '.join([loop_inc] + ustep))
+
+        # Create For header+body
         handle = c.For(loop_init, loop_cond, loop_inc, c.Block(loop_body))
```

Comment:
An UnboundedIndex has no stop condition; in fact, it's unbounded, so no need to change ``loop_cond`` based on the values of the ``uindices``